lib/net.js

```diff
@@ -927,6 +927,8 @@ function lookupAndConnect(self, options) {
   var addressType = exports.isIP(host);
   if (addressType) {
     process.nextTick(function() {
+      if (!self._connecting)
+        return;
```

Comment:
Just an observation. We can actually avoid the empty statement here if we flip the condition